Feature/unit testing

BI003 opened the pull request.

BI003 committed:
В папку Models добавлен класс “Account”, со свойствами Currency, Amount.
В “TestDataGenerator” создан метод, генерирующий словарь, где в качестве ключа находятся клиенты, в качестве значения их банковский счет.
В папку “Application” добавлен проект тестов
“BancSystem.App.Tests”, в котором создан класс “EquivalenceTests”.

BI003 committed:
В классе “EquivalenceTests”, создан метод
“GetHashCodeNecessityPositivTest” в рамках которого получаем словарь тестовых сущностей применяя метод "GenerateDictionaryClientsAndAccounts".
Создаём нового клиента, значения свойств которого,
соответствуют значениям одного из клиентов содержащихся в словаре.
Пробуем получить счет клиента применив созданного
клиента в качестве ключа.

BI003 committed:
Переопределил методы Equals и GetHashCode в классе "Client", и повторил тест.
Работу оформил в виде теста, с проверкой соответствия
результата ожиданию.

BI003 committed:
Усложнили метод "GenerateDictionaryClientsAndAccounts" на случай, если у клиентов окажется несколько банковских счетов.

BI003 committed:
Реализован тест, аналогичный тесту "GetHashCodeNecessityPositivTest", но для списка - List<Emloyee>.

PetrovskiAlex commented on BankSystem.Domain/Models/Client.cs:
```diff
@@ -2,6 +2,20 @@
-       public int Age { get; set; }
+        public int Age { get; set; }
+
+        public override bool Equals(object obj)
```
как то мало полей для эквивалентности клиентов

все Ивановы Иваны эти одни и те же люди?

PetrovskiAlex commented on BankSystem.Domain/Models/Employee.cs:
```diff
@@ -4,5 +4,19 @@ public class Employee : Person
         public string Contract { get; set; }
         public int Salary { get; set; }
+
+        public override bool Equals(object obj)
```
аналогичный комментарий что и для клиентов

PetrovskiAlex commented on BankSystem.Domain/Models/Account.cs:
```diff
@@ -0,0 +1,14 @@
+{
+    public class Account
+    {
+        public string Currency { get; set; }
```
переделать на отдельный тип Currency

PetrovskiAlex commented on BankSystem.App/Services/TestDataGenerator.cs:
```diff
@@ -48,4 +48,54 @@ public Employee FindEmployeeWithMinSalary(List<Employee> employees)
+        while (employees.Count < 1000)
+        {
+            var employee = faker.Generate();
+            if (!employees.Any(e => e.Name == employee.Name && e.Surname == employee.Surname && e.Contract == employee.Contract))
```
зачем же мы переопределяем Equals если потом всё-равно пишем такой код?

PetrovskiAlex commented on BancSystem.App.Tests/EquivalenceTests.cs:
```diff
@@ -0,0 +1,50 @@
+
+            var existingClient = new Client()
+            {
+                Name = clientAndAccount.Keys.First().Name,
```
один раз стоило бы получить clientAndAccount.Keys.First() и взять у него Name и Surname

BI003 committed:
Изменил получение данных для нового клиента, теперь получаю все данные с первого клиента, через: clientAndAccount.Keys.First().
Используем переопределённый метод Equals для сравнения плей.
Добавил поля в классе Employee для более наглядной проверки эквивалентности клиентов.
Добавил поля в классе Client для более наглядной проверки эквивалентности клиентов.

BI003 committed:
В классе Account поменял тип Currency со string на Currency
